Sources/StreamVideo/WebRTC/v2/PeerConnection/Adapters/ICEAdapter.swift

```diff
@@ -252,11 +264,13 @@ actor ICEAdapter: @unchecked Sendable {
             .sinkTask(storeIn: disposableBag) { [weak self] in await self?.trickle($0.candidate) }
             .store(in: disposableBag)
 
-        peerConnection
-            .publisher(eventType: StreamRTCPeerConnection.HasRemoteDescription.self)
-            .log(.debug, subsystems: .iceAdapter)
-            .sinkTask(storeIn: disposableBag) { [weak self] _ in await self?.addTrickledICECandidates() }
-            .store(in: disposableBag)
+        if peerType == .publisher {
```

Comment:
why do we add this check?

Reply:
From the discussion with Marcelo i understood that the the order of applying the candidates matters (needs to happen before creatingAnswer). However upon checking further that doesn't seem to be the case and it's alright to happen at the same time. So removed the check and manual call and now rely on the automatic appliance